hudi-spark-datasource/hudi-spark3-common/src/main/scala/org/apache/spark/sql/hudi/command/AlterTableCommand.scala

```diff
@@ -265,7 +265,7 @@ object AlterTableCommand extends Logging {
 
     val commitActionType = CommitUtils.getCommitActionType(WriteOperationType.ALTER_SCHEMA, metaClient.getTableType)
     val instantTime = client.createNewInstantTime()
-    client.startCommitWithTime(instantTime, commitActionType)
+    client.startAlterTableWithTime(instantTime, commitActionType)
```

Comment:
There is no need to introduce new APIs in base write client, just use `startCommit` should be fine. And it's not for clean but for rollback of failed writes.

Usually we have concurrent modifications to one table tween Flink pipeline and this alter table job, one solution is to config the rollback of failed writes as lazy, but fix here to remove the rollback should also be feasible.

Reply:
> There is no need to introduce new APIs in base write client, just use `startCommit` should be fine. And it's not for clean but for rollback of failed writes.
> 
> Usually we have concurrent modifications to one table tween Flink pipeline and this alter table job, one solution is to config the rollback of failed writes as lazy, but fix here to remove the rollback should also be feasible.

![image](https://github.com/user-attachments/assets/c10493a1-73f8-42ce-8dd0-e07cc40ed23d)

Alter table will involve rollback, clean, and archive,  modify the paras will reduce the intrusiveness.